wallet/walletrpc.c

```diff
@@ -227,23 +252,24 @@ static struct command_result *json_listaddrs(struct command *cmd,
 		}
 
 		// outputs
-		json_object_start(response, NULL);
-		json_add_u64(response, "keyidx", keyidx);
-		json_add_pubkey(response, "pubkey", &pubkey);
-		json_add_string(response, "bech32", out_p2wpkh);
-		json_add_hex_talarr(response, "bech32_redeemscript",
-				    redeemscript_p2wpkh);
-		json_add_string(response, "p2tr", out_p2tr);
-		json_object_end(response);
+		if (address) {
+			char *address_str = encode_scriptpubkey_to_addr(tmpctx, chainparams, address);
+			if (streq(address_str, out_p2wpkh) || streq(address_str, out_p2tr)) {
```

Comment:
Convert address *once*, and since it's actually a scriptpubkey not an address, rename the parameter, too.

Reply:
Fixed, thanks!